### ISSUE TITLE
BUG: Index string ops ignores the order

### ISSUE THREAD

**Issue by sinhrks** [pydata/pandas#10083]
#### Series (OK)

```
s = pd.Series(['A', 'B'])
'x' + s
# 0    xA
# 1    xB
# dtype: object

s + 'y'
# 0    Ay
# 1    By
# dtype: object
```

#### Index (NG)

``` 
idx = pd.Index(['A', 'B'])

# NG
'x' + idx
# Index([u'Ax', u'Bx'], dtype='object')

# OK
idx + 'y'
# Index([u'Ay', u'By'], dtype='object')

**Comment by shoyer** [comment]
@mortada has a fix for this here: https://github.com/pydata/pandas/pull/10021

**Comment by jreback** [comment]
closed by #10021

**Comment by sinhrks** [comment]
Great! I've missed the PR...
